### ISSUE TITLE
感谢分享，请问网盘给出的model和deploy不能加载成功是什么原因？

### ISSUE THREAD

**Issue by zyc4me** [lyxok1/Tiny-DSOD#1]
错误如下

Check failed: target_blobs.size() == source_layer.blobs_size() (1 vs. 2) Incompatible number of blobs for layer Convolution1

我个人认为是bias设置的问题，权重文件里有的层有bias但是deploy设置了false，我一个一个修改后，检测结果一直为空

**Comment by lyxok1** [comment]
the finally released caffemodel is a version where the parameter in BatchNorm and Scale is merged into Convolutional layers before. The source blob_size becomes 2 since the merging process appending an additional  bias to the Convolution layer, while the deploy file is the orignial version for training where BatchNorm and Scale are reserved and the bias for Convolution layers are all false. We are sorry that this is a carelessness during our releasing process. We will either provide an original caffemodel without merging BN or a merging script for our caffemodel  and prototxt later on.

**Comment by liuhengli** [comment]
@zyc4me I also have this error, you have solved.

**Comment by zyc4me** [comment]
@liuhengli I did not solved it.  i try to get prototxt from caffemodel, there is layer type 'deconvoluton' not 'upsample', so the model i think is a lillte old, the model in baidu pan data is MAY 10th, so wait for the new model.

**Comment by liuhengli** [comment]
@lyxok1 OK, Thanks. I think this line needs to be deleted in 'video_detection_demo.py' file，‘ transformer.set_channel_swap('data', (2,1,0))  # the reference model has channels in BGR order instead of RGB’. opencv read video is BGR

**Comment by tf24-karatzhong** [comment]
@lyxok1 Thank you for the pretrianed model and deploy file. When i tried to use pretrained model, I got the following error:[libprotobuf ERROR google/protobuf/text_format.cc:245] Error parsing text-format caffe.NetParameter: 4291:9: Message type "caffe.PriorBoxParameter" has no field named "step".
WARNING: Logging before InitGoogleLogging() is written to STDERR
Is this normal ? i complied this version of caffe successfully.
